duckduckgo/macos-browser
# Sparkle framework & Automatic Updates

Task/Issue URL: https://app.asana.com/0/72649045549333/1200122160553587/f
Tech Design URL: https://app.asana.com/0/481882893211075/1200404130000603/f

**Description**:
Integration of the Sparkle framework for automatic updates.

**Steps to test this PR**:


Manual Update Test 
1. Archive the app
2. Install
3. Run the installed app
4. Click on the menu item DuckDuckGo Privacy Browser -> Check for Updates...
5. Update, follow steps
6. Make sure the browser was updated


Periodic Check Test 
@mallexxx, if there is a better way how to test it, feel free to adjust.
1. In UpdateController.swift, remove the line setting `automaticallyChecksForUpdates` and set `updateCheckInterval` to 1
_Note: [this has a minimum bound of 1 hour in order to keep you from accidentally overloading your servers.](https://sparkle-project.org/documentation/customization/)_ 
🤷‍♂️ 
2. Run the app from Xcode
3. Quit
4. Run the app from Xcode
_Note: Sparkle doesn’t ask the user about updates until the second launch, in order to make your users’ first-launch impression cleaner._
5. Wait for the automatic update dialog (Netflix in the meantime? :D)
6. Cancel the dialog


---
###### Internal references:
[Software Engineering Expectations](https://app.asana.com/0/59792373528535/199064865822552)
[Technical Design Template](https://app.asana.com/0/59792373528535/184709971311943)
**When ready for review, remember to post the PR in MM**
